AnalyticsKitDebugProvider.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import UIKit
+
+class AnalyticsKitDebugProvider: NSObject, AnalyticsKitProvider {
+
+    private weak var alertController: UIAlertController?
+
+    // Lifecycle
+    func applicationWillEnterForeground() { }
+    func applicationDidEnterBackground() { }
+    func applicationWillTerminate() { }
+    func uncaughtException(exception: NSException) { }
```

Comment:
What do you think about putting these no-ops in a protocol extension, so a provider only has to implement the ones relevant to it?
```
extension NSObject: AnalyticsKitProvider {
    func applicationWillEnterForeground(){}
    func applicationDidEnterBackground(){}
    func applicationWillTerminate(){}
    func uncaughtException(exception: NSException){}
    func logScreen(screenName: String){}
    func logEvent(event: String){}
    func logEvent(event: String, withProperty key: String, andValue value: String){}
    func logEvent(event: String, withProperties dict: [String: AnyObject]){}
    func logEvent(event: String, timed: Bool){}
    func logEvent(event: String, withProperties dict: [String: AnyObject], timed: Bool){}
    func endTimedEvent(event: String, withProperties dict: [String: AnyObject]){}
    func logError(name: String, message: String?, exception: NSException?){}
    func logError(name: String, message: String?, error: NSError?){}
}
```